app/public/views/periods/overview/dataExport.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { h, iconChevronBottom } from '/js/src/index.js';
+
+export const exportFormats = {
+    csv: 'CSV',
+    json: 'JSON',
+};
+
+export const exportPreferences = {
+    all: 'all',
+    currentPage: 'currentPage',
+    selected: 'selected',
+    notSelected: 'notSelected',
+    visible: 'visible',
+};
+
+const rowsPreferenceSelectId = 'rows-preference-selection';
+const columnsPreferenceSelectId = 'columns-preference-selection';
+const exportFormatSelectId = 'export-format-selection';
+
+const exportData = async (dataModel, close) => {
+    const remoteData = dataModel.currentPageData;
+
+    remoteData.match({
+        NotAsked: () => {},
+        Loading: () => {},
+        Success: async () => {
+            await dataModel.createDataExport(remoteData.payload, selectedDataFormat());
+        },
+        Failure: () => {},
+    });
+    close();
+};
+
+const handleRowsPreferenceSelection = () => {
+    const exportPreferenceSelection = document.getElementById(rowsPreferenceSelectId);
+    const selectedPreference = exportPreferenceSelection.options[exportPreferenceSelection.selectedIndex].value;
+    switch (selectedPreference) {
+        case exportPreferences.all:
+            /* */
+            break;
+        case exportPreferences.currentPage:
+            /* */
+            break;
+        case exportPreferences.selected:
+            /* */
+            break;
+        case exportPreferences.notSelected:
+            /* */
+            break;
+        case exportPreferences.visible:
+            /* */
+            break;
+        default:
+            break;
+    }
+};
```

Comment:
dead code

Reply:
Right, that should come in the next PR when the handling functions are implemented